ADJ-Internship/BusinessService/Implementations/PurchaseOrderService.cs

```diff
@@ -17,10 +18,233 @@ namespace ADJ.BusinessService.Implementations
+
+                var temp = entity.RowVersion;
+                entity = Mapper.Map(order, entity);
+                entity.RowVersion = temp;
+
+                //foreach (var item in entity.orderDetails)
+                //{
+                //    _orderDetailRepository.Delete(item);
+                //}
+
+                //entity = Mapper.Map<Order>(order);
+                _orderRepository.Update(entity);
+            }
+            else
+            {
+                entity = Mapper.Map<Order>(order);
+
+                _orderRepository.Insert(entity);
+            }
+
+            await UnitOfWork.SaveChangesAsync();
+
+            var rs = Mapper.Map<OrderDTO>(entity);
+            return rs;
+        }
+
+        public async Task<int> GetLastOrderId()
+        {
+            var orderResult = await _orderDataProvider.ListAsync();
+            List<Order> orders = orderResult.Items;
+
+            return orders[orders.Count - 1].Id;
+        }
+
+        public async Task<OrderDetailDTO> CreateOrUpdateOrderDetailAsync(OrderDetailDTO orderDetail)
+        {
+            OrderDetail entity;
+            if (orderDetail.Id > 0)
+            {
+                entity = await _orderDetailRepository.GetByIdAsync(orderDetail.Id, false);
+                if (entity == null)
+                {
+                    throw new AppException("Purchase Order Detail Not Found");
+                }
+
+                var temp = entity.RowVersion;
+                entity = Mapper.Map(orderDetail, entity);
+                entity.RowVersion = temp;
+
+                _orderDetailRepository.Update(entity);
+            }
+            else
+            {
+                entity = Mapper.Map<OrderDetail>(orderDetail);
+                _orderDetailRepository.Insert(entity);
+            }
+
+            await UnitOfWork.SaveChangesAsync();
+
+            var rs = Mapper.Map<OrderDetailDTO>(entity);
+            return rs;
+        }
+
+        public async void DeleteOrderDetailAsync(PagedListResult<OrderDetailDTO> orderDetailsView, int orderId)
+        {
+            var orderDetailResult = await _orderDetailDataProvider.ListAsync();
+            List<OrderDetail> orderDetails = orderDetailResult.Items;
+            
+
+            bool delete;
+            foreach (var item1 in orderDetails)
+            {
+                if (item1.OrderId == orderId)
+                {
+                    delete = true;
+                    foreach (var item2 in orderDetailsView.Items)
+                    {
+                        if ((item2.Id != 0) && (item1.Id == item2.Id))
+                        {
+                            delete = false;
+                        }
+                    }
+
+                    if (delete)
+                    {
+                        _orderDetailRepository.Delete(item1);
+                    }
+                }
+            }
+        }
+
+
+        public async Task<bool> UniquePONumAsync(string PONumber, int? id)
```

Comment:
what is this method for also?
just keep in mind that we never load the whole database just for getting/checking something